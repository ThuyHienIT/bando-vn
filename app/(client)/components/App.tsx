'use client';

import { ConfigProvider } from 'antd';
import { memo } from 'react';
import { ThemeProvider } from 'styled-components';

import { configProviders } from '../antd-theme';
import { theme } from '../theme';

interface Props {
  children: React.ReactNode;
}
export const App = memo<Props>(function BasicLayout(props) {
  return (
    <ThemeProvider theme={theme}>
      <ConfigProvider {...configProviders}>{props.children}</ConfigProvider>
    </ThemeProvider>
  );
});
