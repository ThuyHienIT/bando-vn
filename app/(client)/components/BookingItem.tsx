'use client';

import { Button, Card, Typography } from 'antd';
import dayjs from 'dayjs';
import Image from 'next/image';
import { useCallback } from 'react';

import { EditOutlined } from '@ant-design/icons';

interface Props {
  data: BookingItem;
  onCancel?(data?: BookingItem): void;
  onEdit?(data?: BookingItem): void;
}

export function BookingItem({ onCancel, onEdit, ...props }: Props) {
  const handleCancel = useCallback(() => {
    onCancel?.(props.data);
  }, [onCancel, props.data]);

  const handleEdit = useCallback(() => {
    onEdit?.(props.data);
  }, [onEdit, props.data]);

  return (
    <>
      <Card
        size="small"
        cover={
          props.data.facility?.thumbnail ? (
            <Image
              alt={props.data.facility.name}
              src={props.data.facility?.thumbnail}
              width={100}
              height={100}
            />
          ) : null
        }
        actions={[
          <Button
            key="details"
            type="text"
            size="small"
            danger
            onClick={handleCancel}
          >
            Cancel
          </Button>,
          <Button
            icon={<EditOutlined />}
            type="text"
            size="small"
            key="edit"
            onClick={handleEdit}
          >
            Edit
          </Button>,
        ]}
      >
        <Typography.Title level={5}>
          {props.data.facility?.name}
        </Typography.Title>
        <Typography.Text ellipsis>{`From: ${dayjs(props.data.from).format(
          'DD/MM/YYYY HH:mm'
        )}`}</Typography.Text>
        <Typography.Text ellipsis>{`To: ${dayjs(props.data.to).format(
          'DD/MM/YYYY HH:mm'
        )}`}</Typography.Text>
      </Card>
    </>
  );
}
