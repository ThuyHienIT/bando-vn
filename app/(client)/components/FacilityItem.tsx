'use client';

import { Button, Card, Typography } from 'antd';
import Image from 'next/image';
import Link from 'next/link';
import styled from 'styled-components';

import { EyeOutlined } from '@ant-design/icons';

const CardStyle = styled(Card)`
  height: 100%;
  display: flex;
  flex-direction: column;

  && .ant-card-actions {
    margin-top: auto;
  }
`;

const ImageStyle = styled(Image)`
  width: auto;
  height: auto;
`;

interface Props {
  data: FacilityItem;
}

export function FacilityItemCmp(props: Props) {
  return (
    <CardStyle
      size="small"
      cover={
        props.data.thumbnail ? (
          <ImageStyle
            alt={props.data.name}
            src={props.data.thumbnail}
            width={240}
            height={100}
          />
        ) : null
      }
      actions={[
        <Link href={`/facility/${props.data.id}`} key="details">
          <Button type="text" size="small" icon={<EyeOutlined />}>
            Details
          </Button>
        </Link>,
      ]}
    >
      <Typography.Title level={4}>{props.data.name}</Typography.Title>
      <Typography.Text ellipsis>{props.data.description}</Typography.Text>
    </CardStyle>
  );
}
