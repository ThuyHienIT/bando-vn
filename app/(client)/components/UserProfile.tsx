'use client';

import { CalendarOutlined, ProfileOutlined, UserOutlined } from '@ant-design/icons';
import { Button, Dropdown, Menu, MenuProps } from 'antd';
import Link from 'next/link';
import { memo, useMemo } from 'react';

export const UserProfile = memo(function UserProfile() {
  const items = useMemo<MenuProps['items']>(
    () => [
      {
        label: <Link href="/user/bookings">Your Bookings</Link>,
        key: 'bookings',
        icon: <CalendarOutlined />,
      },
    ],
    []
  );

  return (
    <Dropdown menu={{ items }} placement="bottomRight">
      <Button shape="circle">
        <UserOutlined />
      </Button>
    </Dropdown>
  );
});
