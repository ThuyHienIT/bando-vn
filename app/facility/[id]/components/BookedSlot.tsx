import dayjs from 'dayjs';
import { memo, useCallback } from 'react';
import styled, { css } from 'styled-components';

const SlotStyle = styled.small<{ disabled?: boolean }>`
  background-color: ${(p) => (!p.disabled ? '#509f53' : '#dddddd')};
  color: ${(p) => (!p.disabled ? '#ffffff' : '#333333')};
  border-radius: 2px;
  display: block;
  position: absolute;
  padding: 2px;
  width: 80%;
  z-index: 1;
  ${(p) =>
    !p.disabled &&
    css`
      cursor: pointer;
      &:hover {
        opacity: 0.8;
      }
    `}
`;

interface Props {
  data: BookingItem;
  style?: React.CSSProperties;
  disabled?: boolean;
  onClick?(item: BookingItem): void;
}
export const BookedSlot = memo<Props>(function BookedSlot({
  onClick,
  ...props
}) {
  const fromDayjs = dayjs(props.data.from);
  const toDayjs = dayjs(props.data.to);

  const handleClick = useCallback<React.MouseEventHandler>(
    (e) => {
      e.stopPropagation();
      onClick?.(props.data);
    },
    [onClick, props.data]
  );

  return (
    <SlotStyle
      key={props.data.id}
      style={{
        top: timeToPosition(fromDayjs.format('HH:mm'), 40),
        height: durationToHeight(toDayjs.diff(fromDayjs, 'minute')),
      }}
      disabled={props.disabled}
      onClick={handleClick}
    >
      Booked
    </SlotStyle>
  );
});

/**
 * Convert position in pixels to time
 * @param pos number position of mouse
 * @param gap hour gap in pixels
 * @returns [hour: number, min: number]
 */
export function positionToTime(pos: number, hourGap: number = 40) {
  const val = pos / hourGap;
  const hour = Math.floor(val);
  return [hour, val - hour >= 0.5 ? 30 : 0];
}

/**
 * Convert Time to position
 * @param time string [hh:mm]
 * @param hourGap: number in pixels
 */
export function timeToPosition(time: string, hourGap: number = 40) {
  const [hour, min] = time.split(':').map((i) => parseInt(i, 10));
  return hour * hourGap + (min / 60) * hourGap;
}

/**
 *
 * @param duration number in minutes
 * @param hourGap number pixels per hour
 */
export function durationToHeight(duration: number, hourGap: number = 40) {
  return (duration * hourGap) / 60;
}
