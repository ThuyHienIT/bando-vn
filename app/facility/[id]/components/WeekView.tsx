'use client';

import dayjs, { Dayjs } from 'dayjs';
import { useCallback, useMemo, useState } from 'react';
import styled from 'styled-components';

import { BookingModal } from './BookingModal';
import { DayHeadingCol } from './DayHeadingCol';
import { DaySelectionCol } from './DaySelectionCol';
import { LineCol } from './LineCol';
import { TimeCol } from './TimeCol';

const DayHeadingColStyle = styled(DayHeadingCol)``;
const DaySelectionColStyle = styled(DaySelectionCol)`
  &:first-child {
    border-left: 1px solid #dddddd;
  }
`;
const Wrapper = styled.div`
  display: flex;
  gap: 4px;

  ${DaySelectionColStyle},
  ${DayHeadingColStyle} {
    flex: 1 1 0%;
  }
`;

const TimeColStyle = styled.small`
  > span {
    visibility: hidden;
  }
`;

const DateSelectionScroller = styled.div`
  overflow-y: auto;
  flex-grow: 1;
`;

const DateSelectionContent = styled.div`
  position: relative;
  flex-grow: 1;
`;

const DateSelection = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
`;

const TimeDatesContainer = styled.div`
  display: flex;
`;

interface Props {
  data: FacilityItem;
  selectedDate?: Dayjs;
  occupiedSlots?: BookingItem[];
  onBooked?(item: BookingItem): void;
  onCancalled?(bookingId: string): void;
}

export function WeekView({ onBooked, onCancalled, ...props }: Props) {
  const [opened, setOpened] = useState(false);
  const [from, setFrom] = useState<Dayjs>();
  const [to, setTo] = useState<Dayjs>();
  const [editingBooking, setEditingBooking] = useState<BookingItem>();

  const handleClose = useCallback(() => {
    setOpened(false);
    setFrom(undefined);
    setTo(undefined);
  }, []);

  const days = useMemo(() => {
    const today = props.selectedDate || dayjs();
    const firstDate = today.date() - today.day();

    return Array.from({ length: 7 }).map((_, idx) =>
      today.clone().set('date', firstDate + idx)
    );
  }, [props.selectedDate]);

  const handleSlotClick = useCallback(([from, to]: [Dayjs, Dayjs]) => {
    setFrom(from);
    setTo(to);
    setOpened(true);
  }, []);

  const handleBooked = useCallback(
    (item: BookingItem) => {
      onBooked?.(item);
      setOpened(false);
      setEditingBooking(undefined);
      setFrom(undefined);
      setTo(undefined);
    },
    [onBooked]
  );

  const handleEdit = useCallback((booking: BookingItem) => {
    setEditingBooking(booking);
    setFrom(dayjs(booking.from));
    setTo(dayjs(booking.to));
    setOpened(true);
  }, []);

  const handleCancelled = useCallback(
    (bookingId: string) => {
      onCancalled?.(bookingId);
      setOpened(false);
      setEditingBooking(undefined);
      setFrom(undefined);
      setTo(undefined);
    },
    [onCancalled]
  );

  return (
    <>
      <Wrapper>
        <TimeColStyle>
          <span>00:00</span>
        </TimeColStyle>
        {days.map((d) => (
          <DayHeadingColStyle
            key={d.toString()}
            day={d.format('ddd')}
            date={d.date()}
          />
        ))}
      </Wrapper>

      <DateSelectionScroller id="js-weekview-container" data-testid="scroller">
        <TimeDatesContainer>
          <TimeCol />
          <DateSelectionContent>
            <LineCol />
            <DateSelection>
              {days.map((d) => (
                <DaySelectionColStyle
                  key={d.toString()}
                  date={d.clone()}
                  disabled={d.isBefore(dayjs(), 'date')}
                  onClick={handleSlotClick}
                  onEdit={handleEdit}
                  occupiedSlots={props.occupiedSlots}
                  operationHours={props.data.operationHours}
                />
              ))}
            </DateSelection>
          </DateSelectionContent>
        </TimeDatesContainer>
      </DateSelectionScroller>

      <BookingModal
        opened={opened}
        onClose={handleClose}
        from={from}
        to={to}
        data={props.data}
        onDone={handleBooked}
        bookingId={editingBooking?.id}
        isUpdate={Boolean(editingBooking?.id)}
        onCancelled={handleCancelled}
      />
    </>
  );
}
