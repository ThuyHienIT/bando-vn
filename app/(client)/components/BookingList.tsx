'use client';

import { Col, Empty, Row, Typography } from 'antd';
import Link from 'next/link';
import { useMemo } from 'react';
import styled from 'styled-components';

import { DoubleRightOutlined } from '@ant-design/icons';

import { BookingItem } from './BookingItem';

const ColStyle = styled(Col)``;

const ListStyle = styled(Row)`
  ${ColStyle} {
    margin-bottom: 16px;
  }
`;

const LinkRow = styled.div`
  text-align: right;
`;

interface Props {
  data: BookingItem[];
  heading: string;
  max?: number;
  viewAllLink?: string;
  className?: string;
  onCancel?(item: BookingItem): void;
  onEdit?(item: BookingItem): void;
}

export function BookingList(props: Props) {
  const listData = useMemo(() => {
    if (!props.max || !props.data || props.data.length === 0) return props.data;

    return props.data.slice(0, props.max);
  }, [props.data, props.max]);

  return (
    <div className={props.className}>
      <Typography.Title level={4}>{props.heading}</Typography.Title>
      {listData.length === 0 && (
        <Empty description={`You haven't booked any ${props.heading}`} />
      )}
      {listData.length > 0 && (
        <ListStyle gutter={16}>
          {listData.map((item) => (
            <ColStyle key={item.id} md={6}>
              <BookingItem
                data={item}
                onCancel={props.onCancel}
                onEdit={props.onEdit}
              />
            </ColStyle>
          ))}
        </ListStyle>
      )}

      {props.max && props.data.length > props.max && props.viewAllLink && (
        <LinkRow>
          <Link data-testid="view-all" href={props.viewAllLink}>
            View all <DoubleRightOutlined />
          </Link>
        </LinkRow>
      )}
    </div>
  );
}
