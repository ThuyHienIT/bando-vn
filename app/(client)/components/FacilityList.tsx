'use client';

import Image from 'next/image';
// import { memo } from 'react';
// import { Card, Typography } from 'antd';
import { FacilityItemCmp } from '@app/app/(client)/components/FacilityItem';
import { Col, Row, Typography } from 'antd';
import styled from 'styled-components';
import Link from 'next/link';
import { useMemo } from 'react';
import { DoubleRightOutlined } from '@ant-design/icons';

const ColStyle = styled(Col)``;

const ListStyle = styled(Row)`
  ${ColStyle} {
    margin-bottom: 16px;
  }
`;

const LinkRow = styled.div`
  text-align: right;
`;

interface Props {
  data: FacilityItem[];
  heading: string;
  max?: number;
  viewAllLink?: string;
  className?: string;
}

export function FacilityList(props: Props) {
  const listData = useMemo(() => {
    if (!props.max) return props.data;

    return props.data.slice(0, props.max);
  }, [props.data, props.max]);

  return (
    <div className={props.className}>
      <Typography.Title level={4}>{props.heading}</Typography.Title>
      <ListStyle gutter={16}>
        {listData.map((item) => (
          <ColStyle key={item.id} md={6}>
            <FacilityItemCmp data={item} />
          </ColStyle>
        ))}
      </ListStyle>
      {props.max && props.data.length > props.max && props.viewAllLink && (
        <LinkRow>
          <Link href={props.viewAllLink}>
            View all <DoubleRightOutlined />
          </Link>
        </LinkRow>
      )}
    </div>
  );
}
