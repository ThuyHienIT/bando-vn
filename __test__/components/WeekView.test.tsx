import '@testing-library/jest-dom';

import {
  generateBooking,
  generateFacility,
  userInitialState,
} from '__test__/helpers';
import { WeekView } from 'app/facility/[id]/components/WeekView';
import dayjs from 'dayjs';
import { RecoilRoot } from 'recoil';

import { FacilityTypeEnum } from '@enums';
import { fireEvent, render, screen } from '@testing-library/react';

describe('WeekView List', () => {
  it('smapshot', () => {
    const fac = generateFacility(FacilityTypeEnum.Room);
    fac.id = 'test';
    const { container } = render(
      <RecoilRoot initializeState={userInitialState}>
        <WeekView
          data={fac}
          selectedDate={dayjs()}
          occupiedSlots={[generateBooking(fac.id)]}
        />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });

  it('verify elements', async () => {
    const fac = generateFacility(FacilityTypeEnum.Room);
    fac.id = 'test';
    const today = dayjs();

    const { container } = render(
      <RecoilRoot initializeState={userInitialState}>
        <WeekView data={fac} selectedDate={today} />
      </RecoilRoot>
    );

    const scrollEl = screen.getByTestId('scroller');

    const clickToBookEl = screen.getAllByTestId('click-to-book');

    fireEvent(
      clickToBookEl[today.day() + 1],
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      })
    );

    const bookingFormEl = screen.getByTestId('booking-form');
    expect(bookingFormEl).toBeInTheDocument();

    expect(scrollEl).toBeInTheDocument();
    expect(scrollEl.getAttribute('id')).toBe('js-weekview-container');
  });

  it('has booked slots', async () => {
    const fac = generateFacility(FacilityTypeEnum.Room);
    fac.id = 'test';
    const booking = generateBooking(FacilityTypeEnum.Room);
    booking.from = dayjs().format();
    booking.to = dayjs().add(30, 'minute').format();

    render(
      <RecoilRoot initializeState={userInitialState}>
        <WeekView
          data={fac}
          selectedDate={undefined}
          occupiedSlots={[booking]}
        />
      </RecoilRoot>
    );

    const bookedEl = await screen.findByText('Booked');
    expect(bookedEl).toBeInTheDocument();

    fireEvent.click(bookedEl);

    const bookingFormEl = await screen.findByTestId('booking-form');
    expect(bookingFormEl).toBeInTheDocument();
  });
});
