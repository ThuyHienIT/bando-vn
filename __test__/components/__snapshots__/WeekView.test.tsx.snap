// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekView List smapshot 1`] = `
.c3 {
  text-transform: uppercase;
}

.c4 {
  font-size: 40px;
  font-weight: 400;
  margin-bottom: 8px;
}

.c2 {
  text-align: center;
  color: #333;
}

.c12 {
  width: 80px;
  min-width: 80px;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
  border-right: 1px solid #dddddd;
  position: relative;
  background-color: rgba(0,0,0,0.05);
}

.c14 {
  width: 80px;
  min-width: 80px;
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
  border-right: 1px solid #dddddd;
  position: relative;
}

.c15 {
  background-color: rgba(0,0,0,0.05);
  display: block;
  position: absolute;
  width: 100%;
}

.c10 {
  border-top: 1px solid #dddddd;
  height: 40px;
}

.c10:last-child {
  border-bottom: 1px solid #dddddd;
}

.c7 {
  height: 40px;
}

.c8 {
  color: #8d8d8d;
  display: inline-block;
  position: relative;
  top: -10px;
  background: #fff;
  padding-right: 4px;
}

.c13:first-child {
  border-left: 1px solid #dddddd;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
}

.c1 > span {
  visibility: hidden;
}

.c5 {
  overflow-y: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c9 {
  position: relative;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c11 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <small
      class="c1"
    >
      <span>
        00:00
      </span>
    </small>
    <div
      class="c2 "
    >
      <span
        class="ant-typography ant-typography-secondary c3 css-dev-only-do-not-override-diro6f"
      >
        Sun
      </span>
      <h2
        class="c4"
      >
        19
      </h2>
    </div>
    <div
      class="c2 "
    >
      <span
        class="ant-typography ant-typography-secondary c3 css-dev-only-do-not-override-diro6f"
      >
        Mon
      </span>
      <h2
        class="c4"
      >
        20
      </h2>
    </div>
    <div
      class="c2 "
    >
      <span
        class="ant-typography ant-typography-secondary c3 css-dev-only-do-not-override-diro6f"
      >
        Tue
      </span>
      <h2
        class="c4"
      >
        21
      </h2>
    </div>
    <div
      class="c2 "
    >
      <span
        class="ant-typography ant-typography-secondary c3 css-dev-only-do-not-override-diro6f"
      >
        Wed
      </span>
      <h2
        class="c4"
      >
        22
      </h2>
    </div>
    <div
      class="c2 "
    >
      <span
        class="ant-typography ant-typography-secondary c3 css-dev-only-do-not-override-diro6f"
      >
        Thu
      </span>
      <h2
        class="c4"
      >
        23
      </h2>
    </div>
    <div
      class="c2 "
    >
      <span
        class="ant-typography ant-typography-secondary c3 css-dev-only-do-not-override-diro6f"
      >
        Fri
      </span>
      <h2
        class="c4"
      >
        24
      </h2>
    </div>
    <div
      class="c2 "
    >
      <span
        class="ant-typography ant-typography-secondary c3 css-dev-only-do-not-override-diro6f"
      >
        Sat
      </span>
      <h2
        class="c4"
      >
        25
      </h2>
    </div>
  </div>
  <div
    class="c5"
    data-testid="scroller"
    id="js-weekview-container"
  >
    <div
      class="c6"
    >
      <div>
        <div
          class="c7"
        >
          <small
            class="c8"
          />
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            01:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            02:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            03:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            04:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            05:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            06:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            07:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            08:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            09:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            10:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            11:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            12:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            13:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            14:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            15:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            16:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            17:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            18:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            19:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            20:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            21:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            22:00
          </small>
        </div>
        <div
          class="c7"
        >
          <small
            class="c8"
          >
            23:00
          </small>
        </div>
      </div>
      <div
        class="c9"
      >
        <div>
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
          <div
            class="c10"
          />
        </div>
        <div
          class="c11"
        >
          <div
            class="c12 c13"
            data-testid="click-to-book"
          >
            <div
              role="presentation"
            />
          </div>
          <div
            class="c12 c13"
            data-testid="click-to-book"
          >
            <div
              role="presentation"
            />
          </div>
          <div
            class="c12 c13"
            data-testid="click-to-book"
          >
            <div
              role="presentation"
            />
          </div>
          <div
            class="c12 c13"
            data-testid="click-to-book"
          >
            <div
              role="presentation"
            />
          </div>
          <div
            class="c12 c13"
            data-testid="click-to-book"
          >
            <div
              role="presentation"
            />
          </div>
          <div
            class="c14 c13"
            data-testid="click-to-book"
          >
            <div
              role="presentation"
            >
              <small
                class="c15"
                style="top: 0px; height: 320px;"
              />
              <small
                class="c15"
                style="top: 760px; height: 199.33333333333334px;"
              />
            </div>
          </div>
          <div
            class="c14 c13"
            data-testid="click-to-book"
          >
            <div
              role="presentation"
            >
              <small
                class="c15"
                style="top: 0px; height: 320px;"
              />
              <small
                class="c15"
                style="top: 760px; height: 199.33333333333334px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
